### PR TITLE
AUTH-77_Infusionsoft_Id_Can_Be_Changed_To_An_Invalid_Email_Address

### PR DESCRIPTION
**Pull request by travisbrkr1234**
Added regex from jquery.validate.js to User to match the front-end validation

### CODE REVIEW

File: cas/src/main/resources/custom_messages.properties [hunk 1]
```diff
@@ -33,7 +33,7 @@ login.noUsername=Email address is required.
 login.noPassword=Password is required.
 
 # Text for the registration page
-registration.error.exception=Couldn't create your Infusionsoft ID. Please check you are using a valid email address that's not already in use.
+registration.error.exception=Couldn't create your Infusionsoft ID. Please check to be sure you are using a valid email address (e.g. email@example.com) that's not already in use..
```
Comment (bbooth) [inline cas/src/main/resources/custom_messages.properties]: extra period..

File: cas/src/test/java/org/apereo/cas/infusionsoft/services/UserEmailValidationTest.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                "abc@insta-.com",
+                "ABC.ABC.ABC.ABC0046@domain.com.com.com",
+                "this@that@this.com");
+
+        invalidEmailAddresses.forEach(this::validateInvalidEmailAddress);
+    }
+
+    private void validateValidEmailAddress(String emailAddress) {
+        boolean isValid = emailAddress.matches(user.getEmailRegex());
+        Assert.assertTrue("This email should be considered invalid: " + emailAddress, isValid);
```
Comment (ntrecina) [inline cas/src/test/java/org/apereo/cas/infusionsoft/services/UserEmailValidationTest.java]: It seems like the assert message should "This email should be considered valid:".

File: cas/src/test/java/org/apereo/cas/infusionsoft/services/UserEmailValidationTest.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    private void validateInvalidEmailAddress(String emailAddress) {
+        boolean isValid = emailAddress.matches(user.getEmailRegex());
+        Assert.assertTrue("This email should be considered valid: " + emailAddress, !isValid);
```
Comment (ntrecina) [inline cas/src/test/java/org/apereo/cas/infusionsoft/services/UserEmailValidationTest.java]: It seems like the assert message should "This email should be considered invalid:".

File: cas/src/test/java/org/apereo/cas/infusionsoft/services/UserEmailValidationTest.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                "abc123..45@insta.com",
+                "abc@insta-.com",
+                "ABC.ABC.ABC.ABC0046@domain.com.com.com",
+                "this@that@this.com");
+
+        invalidEmailAddresses.forEach(this::validateInvalidEmailAddress);
+    }
+
+    private void validateValidEmailAddress(String emailAddress) {
+        boolean isValid = emailAddress.matches(user.getEmailRegex());
```
Comment (MikeChristianson) [inline cas/src/test/java/org/apereo/cas/infusionsoft/services/UserEmailValidationTest.java]: This test should use the same case sensitivity as the [actual implementation](https://github.com/infusionsoft/cas-gradle-overlay-template/blob/9b1332c5053e85f94c55b797f721815c6711b58f/cas/src/main/java/org/apereo/cas/infusionsoft/domain/User.java#L79).

File: cas/src/test/java/org/apereo/cas/infusionsoft/services/UserEmailValidationTest.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        Assert.assertTrue("This email should be considered invalid: " + emailAddress, isValid);
+    }
+
+    private void validateInvalidEmailAddress(String emailAddress) {
+        boolean isValid = emailAddress.matches(user.getEmailRegex());
```
Comment (MikeChristianson) [inline cas/src/test/java/org/apereo/cas/infusionsoft/services/UserEmailValidationTest.java]: This test should use the same case sensitivity as the [actual implementation](https://github.com/infusionsoft/cas-gradle-overlay-template/blob/9b1332c5053e85f94c55b797f721815c6711b58f/cas/src/main/java/org/apereo/cas/infusionsoft/domain/User.java#L79).

File: cas/src/test/java/org/apereo/cas/infusionsoft/services/UserEmailValidationTest.java (hunk 2 above)
Comment (travisbrkr1234) [inline cas/src/test/java/org/apereo/cas/infusionsoft/services/UserEmailValidationTest.java]: This should be ok, the funky thing is that this method is "inverse". The message will be displayed when the assertion fails. For example: putting an invalid email address in the valid collection will result in the error message:
screen shot 2017-08-07 at 2 21 39 pm

File: cas/src/test/java/org/apereo/cas/infusionsoft/services/UserEmailValidationTest.java (hunk 3 above)
Comment (travisbrkr1234) [inline cas/src/test/java/org/apereo/cas/infusionsoft/services/UserEmailValidationTest.java]: Same thing with this method, the message will be displayed on the failure of the assertion:
screen shot 2017-08-07 at 2 22 51 pm

File: cas/src/test/java/org/apereo/cas/infusionsoft/services/UserEmailValidationTest.java (hunk 5 above)
Comment (travisbrkr1234) [inline cas/src/test/java/org/apereo/cas/infusionsoft/services/UserEmailValidationTest.java]: Oops, thanks 👍

File: cas/src/test/java/org/apereo/cas/infusionsoft/services/UserEmailValidationTest.java (hunk 4 above)
Comment (travisbrkr1234) [inline cas/src/test/java/org/apereo/cas/infusionsoft/services/UserEmailValidationTest.java]: Oops, thanks 👍

**Pull request merged**